src/allotropy/parsers/utils/pandas.py

```diff
@@ -88,6 +90,31 @@ def assert_value_from_df(df: pd.DataFrame, key: str) -> Any:
         raise AllotropeConversionError(msg) from e
 
 
+def parse_header_row(df: pd.DataFrame) -> pd.DataFrame:
+    # Set the first row of a dataframe as the columns. This is useful when the dataframe has already been
+    # parsed (e.g. when splitting a dataframe into two parts), and so the header/index_col arguments cannot
+    # be used to set the columns/index at read time.
+    df.columns = pd.Index(
+        assert_not_empty_df(df, "Cannot set parse header row for empty dataframe.")
+        .astype(str)
+        .iloc[0]
+    )
+    df.columns = df.astype(str).columns.str.strip()
+    return df[1:]
+
+
+def split_header_and_data(
+    df: pd.DataFrame, cond: Callable[[pd.Series[Any]], bool]
+) -> tuple[pd.DataFrame, pd.DataFrame]:
+    for idx, row in df.iterrows():
+        if cond(row):
+            header_end = int(str(idx))
+            return df[:header_end], df[header_end + 1 :]
+
+    msg = f"Unable to split header and data from dataframe: {df}"
```

Comment:
putting `df` in the message would put the entire table, right? (I think the actual data is its `__repr__`?) if that's the case should we put something smaller? or like the first few lines or something

Reply:
df truncates the df somewhat, I think in this case it would be good to have as much info as possible. Here's an example:

```
E       ValueError: Here is my error   Sample ID  Cell count  Viable cells  ...  Dilution         Wash  Tag
E       0    CLB001        2191          2123  ...         1  Normal Wash  NaN
E       1    CLB002        7670          7425  ...         1  Normal Wash  NaN
E       2    CLB003        1519          1494  ...         1  Normal Wash  NaN
E       3    CLB005        3195          3137  ...         1  Normal Wash  NaN
E       4    CLB006        2493          2379  ...         1  Normal Wash  NaN
E       5    CLB007        2920          2842  ...         1  Normal Wash  NaN
E       6    CLB008        7001          6896  ...         1  Normal Wash  NaN
E       7    CLB009        5284          5194  ...         1  Normal Wash  NaN
E       8    CLB010       11360          9329  ...         1  Normal Wash  NaN
E       9    CLB011        9799          7061  ...         1  Normal Wash  NaN
E
E       [10 rows x 33 columns]E       ValueError: Here is my error   Sample ID  Cell count  Viable cells  ...  Dilution         Wash  Tag
E       0    CLB001        2191          2123  ...         1  Normal Wash  NaN
E       1    CLB002        7670          7425  ...         1  Normal Wash  NaN
E       2    CLB003        1519          1494  ...         1  Normal Wash  NaN
E       3    CLB005        3195          3137  ...         1  Normal Wash  NaN
E       4    CLB006        2493          2379  ...         1  Normal Wash  NaN
E       5    CLB007        2920          2842  ...         1  Normal Wash  NaN
E       6    CLB008        7001          6896  ...         1  Normal Wash  NaN
E       7    CLB009        5284          5194  ...         1  Normal Wash  NaN
E       8    CLB010       11360          9329  ...         1  Normal Wash  NaN
E       9    CLB011        9799          7061  ...         1  Normal Wash  NaN
E
E       [10 rows x 33 columns]
```